core/include/kernel/pm.h

```diff
@@ -133,6 +133,24 @@ static inline void register_pm_driver_cb(
 						       PM_CB_ORDER_DRIVER));
 }
 
+/*
+ * Register a core service callback for generic suspend/resume.
+ * Refer to struct pm_callback_handle for description of the callbacks
+ * API.
+ *
+ * @callback: Registered callback function
+ * @handle: Registered private handle argument for the callback
+ */
+static inline void register_pm_core_service_cb(
+		TEE_Result (*callback)(
```

Comment:
This screams: give me a typedefed function pointer please!

Reply:
addressed